### PR TITLE
feat: improve async actions docs and guards

### PR DESCRIPTION
**Pull request by coroiu**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

- Improve docs after questions I've gotten on slack
- Improve guards against multiple requests happening at the same time

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: libs/components/src/async-actions/bit-action.directive.ts [hunk 1]
```diff
@@ -39,7 +41,7 @@ export class BitActionDirective implements OnDestroy {
 
   @HostListener("click")
   protected async onClick() {
-    if (!this.handler) {
+    if (!this.handler || this.disabled || this.buttonComponent.disabled) {
```
Comment (Hinton) [inline libs/components/src/async-actions/bit-action.directive.ts:44]: We don't seem to ever set `this.disabled`.

Comment (coroiu) [reply libs/components/src/async-actions/bit-action.directive.ts:44]: https://github.com/bitwarden/clients/blob/fb534280531ba941a0843e5e8a39d38d6ad219f0/libs/components/src/async-actions/form-button.directive.ts#L54-L56

It's set by the form button directive

Comment (coroiu) [reply libs/components/src/async-actions/bit-action.directive.ts]: fixed!

**Review by Hinton — APPROVED** [review]
Thanks

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Thank you for the clarifications!

**Pull request merged**